acceptance/setup/aio/010_Install.rb

```diff
@@ -29,8 +32,20 @@
 
 step "Install puppetserver..." do
   if ENV['SERVER_VERSION']
-    install_puppetlabs_dev_repo(master, 'puppetserver', ENV['SERVER_VERSION'])
-    install_puppetlabs_dev_repo(master, 'puppet-agent', ENV['SHA'])
+    ENV['AGENT_DOWNLOAD_URL'] ||= DEFAULT_INTERNAL_DOWNLOAD_URL
+    ENV['SERVER_DOWNLOAD_URL'] ||= DEFAULT_NIGHTLIES_DOWNLOAD_URL
+    # We can only use a SHAs of "latest" if we're downloading from nightlies, as
+    # our other options, such as 'builds.delivery.puppetlabs.net', do not have
+    # "latest" directories.
+    if ENV['SHA'] == 'latest' and ENV['AGENT_DOWNLOAD_URL'] != DEFAULT_NIGHTLIES_DOWNLOAD_URL
+      fail "Can only use ENV['SHA'] == 'latest' if ENV['AGENT_DOWNLOAD_URL'] == '#{DEFAULT_NIGHTLIES_DOWNLOAD_URL}'"
+    end
+
+    if ENV['SERVER_VERSION'] == 'latest' and ENV['SERVER_DOWNLOAD_URL'] != DEFAULT_NIGHTLIES_DOWNLOAD_URL
+      fail "Can only use ENV['SERVER_VERSION'] == 'latest' if ENV['SERVER_DOWNLOAD_URL'] == '#{DEFAULT_NIGHTLIES_DOWNLOAD_URL}'"
+    end
+    install_puppetlabs_dev_repo(master, 'puppetserver', ENV['SERVER_VERSION'], nil, :dev_builds_url => ENV['SERVER_DOWNLOAD_URL'])
+    install_puppetlabs_dev_repo(master, 'puppet-agent', ENV['SHA'], nil, :dev_builds_url => ENV['AGENT_DOWNLOAD_URL'])
     master.install_package('puppetserver')
   else
     # beaker can't install puppetserver from nightlies (BKR-673)
```

Comment:
Similar comment as in https://github.com/puppetlabs/puppet/pull/5908#discussion_r118386404